NSS-Day-Cohort-42/rare-media-moguls
# Rp publish

#### Changes Made
1. added more filtering to posts in post list depending on publication date, if user is admin, and if post was authored by user
1. added button on post detail to unpublish/publish a post if it was authored by the user
1. changed dates on post detail and post list to say unpublished if date is null

#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout rp-publish
```
2. Run the application.
```
npm start
```
3. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py server
```
4. Test app functionality.
> go to post detail and click the unpublish/publish buttons
>unpublish a post and go to post list. The post should still appear since the user is the author and the date should say unpublished
>login as another user. The unpublished post should not appear on the post list even for admins. 

5. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
